lib/galaxy/datatypes/sniff.py

```diff
@@ -478,7 +477,7 @@ def handle_uploaded_dataset_file( filename, datatypes_registry, ext='auto', is_m
         ext = guess_ext( filename, sniff_order=datatypes_registry.sniff_order, is_multi_byte=is_multi_byte )
 
     if check_binary( filename ):
-        if not Binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
+        if not galaxy.datatypes.binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
```

Comment:
Compilation breaks when I have both **'from galaxy.datatypes.binary import Binary'** in **sniff.py** and **'from galaxy.datatypes.tabular import Sam'** in **binary.py** . I donot understand the reason behind that. Code is comping properly only when i remove **'from galaxy.datatypes.binary import Binary'**  from **sniff.py**.